server/leafnode.go

```diff
@@ -934,7 +953,13 @@ func (s *Server) initLeafNodeSmap(c *client) {
 	for isubj := range acc.imports.services {
 		ims = append(ims, isubj)
 	}
-	acc.mu.RUnlock()
+	// Create a unique subject that will be used for loop detection.
+	lds := acc.lds
```

Comment:
Why not just `if acc.lds == _EMPTY_ { acc.lds = ... }`?

Reply:
Because I need the subject down at line 1005: `c.leaf.smap[lds]++`, no?

Reply:
Ah ;)